src/pounder/mod.rs

```diff
@@ -0,0 +1,186 @@
+use mcp23017;
+use ad9959;
+
+pub mod error;
+pub mod attenuators;
+mod rf_power;
+pub mod types;
+
+use super::hal;
+
+use error::Error;
+use attenuators::AttenuatorInterface;
+use types::{DdsChannel, InputChannel};
+use rf_power::PowerMeasurementInterface;
+
+use embedded_hal::{
+    blocking::spi::Transfer,
+    adc::OneShot
+};
+
+#[allow(dead_code)]
+const OSC_EN_N_PIN: u8 = 8 + 7;
+
+const EXT_CLK_SEL_PIN: u8 = 8 + 6;
+
+const ATT_RST_N_PIN: u8 = 8 + 5;
+
+const ATT_LE0_PIN: u8 = 8 + 0;
+const ATT_LE1_PIN: u8 = 8 + 1;
+const ATT_LE2_PIN: u8 = 8 + 2;
+const ATT_LE3_PIN: u8 = 8 + 3;
+
+pub struct QspiInterface {
+    pub qspi: hal::qspi::Qspi,
+}
+
+impl ad9959::Interface for QspiInterface {
+    type Error = Error;
+
+    fn configure_mode(&mut self, mode: ad9959::Mode) -> Result<(), Error> {
```

Comment:
Only `FourBit` can be made to work reasonably. No need to bother with the other modes.
c.f.
https://github.com/quartiq/ad9959/commit/33a90e2e271cc4a5d96b2a4d19ea181bfc2be288

Reply:
I've reverted the referenced changes out of the `ad9959` driver and moved them into the Qspi interface implementation in this crate. This keeps the `ad9959` device-independant, since the problems mentioned in those comments is related to the STM32H7 we're using and not the AD9959.

Reply:
I think it is a incompatibility between the two, not solely attributable to either.
But `QspiInterface` is certainly the right spot to address it. Excellent.